sdk/cosmosdb/cosmos/test/public/functional/item.spec.ts

```diff
@@ -711,7 +712,12 @@ describe("bulk/batch item operations", function () {
       assert.strictEqual(deleteResponse.result[1].statusCode, 404);
       const { resource: readItem } = await container.item(otherItemId).read();
       assert.strictEqual(readItem, undefined);
+      assert(isOperationResponse(deleteResponse.result[0]));
     });
+
+    function isOperationResponse(object: any): boolean {
+      return "statusCode" in object && "requestCharge" in object;
+    }
```

Comment:
This alternative is annoyingly more verbose, but a lot more robust. The current version will throw an error instead of returning false if `object` isn't a non-null object.

```suggestion
    function isOperationResponse(object: unknown): object is OperationResponse {
      return typeof object === "object" && object !== null && Object.prototype.hasOwnProperty.call(object, "statusCode") && Object.prototype.hasOwnProperty.call(object, "requestCharge");
    }
```

Reply:
Hey @witemple-msft just curious. Shouldn't adding `object !== null` after `typeof object === 'object'` be enough. Is there any advantage of using `Object.prototype.hasOwnProperty.call(object, "statusCode")` over `'statusCode' in object`.